src/dreyfus_index_updater.erl

```diff
@@ -31,6 +31,14 @@ update(IndexPid, Index) ->
     erlang:put(io_priority, {view_update, DbName, IndexName}),
     {ok, Db} = couch_db:open_int(DbName, []),
     try
+        {ok, IdxPurgeSeq} = clouseau_rpc:get_purge_seq(IndexPid),
+        FoldFun = fun(PurgeSeq, {Id, _Revs}, Acc) ->
+            clouseau_rpc:delete(IndexPid, Id),
```

Comment:
I read couch_index/couch_mrview code and I think it is same to drefus. It is also delete first and then rebuild it.
If we purged a document and then search doc. Here is the steps in dreyfus.
1. fold purged docs and it will delete the indexes of purged docs.
2. fold db changes and it will rebuild the indexes of docs if they are still exist.
3. do search.
The operations are triggered by search request. Delete and update index first and then search and return the result to the caller.